src/packsquash.js

```diff
@@ -0,0 +1,59 @@
+import { endGroup, exportVariable, getBooleanInput, startGroup, warning } from '@actions/core';
+import { exec } from '@actions/exec';
+import { rmSync } from 'fs';
+import { addProblemMatcher, removeProblemMatcher } from './problem-matcher';
+import { Options } from './options';
+
+/**
+ * @param {WorkingDirectory} workingDirectory
+ * @returns {Promise<void>}
+ */
+export async function printPackSquashVersion(workingDirectory) {
+    startGroup('PackSquash version');
+    await exec(workingDirectory.packsquash, ['--version']);
+    endGroup();
+}
+
+/**
+ * @param {WorkingDirectory} workingDirectory
+ * @returns {Promise<void>}
+ */
+export async function runPackSquash(workingDirectory) {
+    async function run(description) {
+        addProblemMatcher(workingDirectory.problemMatcher);
+        if (description) {
+            startGroup(`PackSquash output (${description})`);
+        } else {
+            startGroup('PackSquash output');
+        }
+        const exitCode = await exec(workingDirectory.packsquash, [workingDirectory.options]);
+        endGroup();
+        removeProblemMatcher(workingDirectory.problemMatcher);
+        return exitCode;
+    }
+
+    let exitCode = await run();
+    switch (exitCode) {
+        case 0:
+            break;
+        case 129:
+            warning('PackSquash reported that the previous ZIP file could not be used to speed up processing. Discarding it.');
+            rmSync(workingDirectory.zip);
```

Comment:
I'd make this `async`:

```suggestion
            await rm(workingDirectory.zip);
```

Reply:
Fix by b6b30e7f70e736cf727e199c89fe64280ce26466